### PR TITLE
Add a section about custom user config

### PR DESCRIPTION
**Pull request by awav**
subj.

### CODE REVIEW

File: doc/source/notebooks/intro_to_gpflow2.pct.py [hunk 1]
```diff
@@ -477,3 +477,31 @@ def checkpointing_training_loop(
...
+# %% [markdown]
+# ## User config update
+#
+# In this notebook, we used a lot `gpflow.config` methods for setting and getting default attributes from global configuration. However, GPflow provides a way for local config modification without updating values in global. As you can see below, using `gpflow.config.as_context` replaces temporarily global config with your instance. At creation time, custom config instance uses standard values from the global config:
+
+# %%
+user_config = gpflow.config.Config(float=tf.float32, positive_bijector="exp")
+
+user_str = "User config\t"
+global_str = "Global config\t"
```
Comment (st--) [inline doc/source/notebooks/intro_to_gpflow2.pct.py]: Nit: I got a bit confused why we'd need a \tab in here - I might have moved that into the print() call itself...

File: doc/source/notebooks/intro_to_gpflow2.pct.py [hunk 2]
```diff
@@ -477,3 +477,31 @@ def checkpointing_training_loop(
...
+
+with gpflow.config.as_context(user_config):
+    print(f"{user_str}gpflow.config.default_float = {gpflow.config.default_float()}")
+    print(f"{user_str}gpflow.config.positive_bijector = {gpflow.config.default_positive_bijector()}")
+
+print(f"{global_str}gpflow.config.default_float = {gpflow.config.default_float()}")
+print(f"{global_str}gpflow.config.positive_bijector = {gpflow.config.default_positive_bijector()}")
+
+# %%
+user_config = gpflow.config.Config(float=tf.float32, positive_bijector="exp")
```
Comment (st--) [inline doc/source/notebooks/intro_to_gpflow2.pct.py]: nit: this is the same as in the previous cell, is that intentional?

File: doc/source/notebooks/intro_to_gpflow2.pct.py (hunk 1 above)
Comment (awav) [reply doc/source/notebooks/intro_to_gpflow2.pct.py]: I didn't want to pollute prints with tab symbol

File: doc/source/notebooks/intro_to_gpflow2.pct.py (hunk 2 above)
Comment (awav) [reply doc/source/notebooks/intro_to_gpflow2.pct.py]: Yes, it shows that a user config has global scale impact.

**Pull request merged**